src/Microsoft.Health.Fhir.Shared.Core/Features/Operations/Reindex/ReindexUtilities.cs

```diff
@@ -69,13 +73,7 @@ public async Task ProcessSearchResultsAsync(SearchResult results, IReadOnlyDicti
                 }
 
                 entry.Resource.SearchParameterHash = searchParamHash;
-                var resourceElement = _deserializer.Deserialize(entry.Resource);
-                var newIndices = _searchIndexer.Extract(resourceElement);
-
-                // TODO: If it reasonable to do so, we can compare
-                // old and new search indices to avoid unnecessarily updating search indices
-                // when not changes have been made.
-                entry.Resource.UpdateSearchIndices(newIndices);
+                _resourceWrapperFactory.Update(entry.Resource);
```

Comment:
@feordin, I'd like to move this into the ResourceWrapperFactory to consolidate Sort min/max calculation